test/test_pruning_op.py

```diff
@@ -17,7 +18,7 @@ class PruningOpTest(TestCase):
     # We mask a row if its indicator value is less than the threshold.
     def _generate_rowwise_mask(self, embedding_rows):
         indicator = torch.from_numpy((np.random.random_sample(embedding_rows)).astype(np.float32))
-        threshold = np.random.random_sample()
+        threshold = random.uniform(0, 1)
```

Comment:
weird

Reply:
moving form NumPy scalars (which we treat as 0D tensors) to a regular scalar, as, alas, here we are changing the semantics of this type of code

Reply:
Actually the fix in this cases is simply to do a cast into an actual Python scalar. Changed to that as it's less intrusive.

Now, this function is indeed proof of what humans are capable of writing if you give them access to enough libraries.